components/sup/src/manager/service/config.rs

```diff
@@ -174,11 +175,20 @@ impl Cfg {
             .replace("-", "_");
         match env::var(&var_name) {
             Ok(config) => {
-                let toml = toml::de::from_str(&config).map_err(|e| {
-                    sup_error!(Error::TomlParser(e))
-                })?;
-                self.environment = Some(toml::Value::Table(toml));
-
+                match toml::de::from_str(&config) {
+                    Ok(toml) => self.environment = Some(toml::Value::Table(toml)),
+                    Err(e) => {
+                        outputln!("{}", sup_error!(Error::TomlParser(e)));
```

Comment:
I wouldn't output that there was a parser error here since we're going to try and fallback to `JSON`. We can think of this instead like a "hint" system where we aren't sure what your intent was - we don't want to warn or error because we don't actually know what the intent was.

I would remove the Json/Toml parser errors here entirely and instead use early returns to short circuit the function. If we don't early return out, have the last line of this return an error like `Error::BadEnvConfig` which states that we were unable to read valid TOML or JSON and print out the contents that we did receive. You can store the contents in the enum variant like this `Error::BadEnvConfig(String)`